fileserver/cook/file_server.py

```diff
@@ -0,0 +1,164 @@
+
+
+class FileServerApplication(gunicorn.app.base.BaseApplication):
+
+    def __init__(self, options=None):
+        self.options = options or {}
+        self.application = app
+        super(FileServerApplication, self).__init__()
+
+    def load_config(self):
+        config = dict([(key, value) for key, value in iteritems(self.options)
+                       if key in self.cfg.settings and value is not None])
+        for key, value in iteritems(config):
+            self.cfg.set(key.lower(), value)
+
+    def load(self):
+        return self.application
+
+
+@app.route('/files/download')
+@app.route('/files/download.json')
+def download():
+    path = request.args.get('path')
+    if path is None:
+        return "Expecting 'path=value' in query.\n", 400
+    if not os.path.exists(path):
+        return "", 404
+    if os.path.isdir(path):
+        return "Cannot download a directory.\n", 400
+    return send_file(path, as_attachment=True)
+
+
+def try_parse_int(param_name, val):
+    err_message = lambda: f"Failed to parse {param_name}: Failed to convert '{val}' to number.\n"
+    try:
+        int_val = int(val)
+    except ValueError as _:
+        return (None, err_message())
+    except Exception as _:
+        return (None, err_message())
+    return (int_val, None)
+
+
+@app.route('/files/read')
+@app.route('/files/read.json')
+def read():
+    path = request.args.get('path')
+    offset_param = request.args.get('offset')
+    length_param = request.args.get('length')
+    if path is None:
+        return "Expecting 'path=value' in query.\n", 400
+    offset, err = (None, None) if offset_param is None else try_parse_int("offset", offset_param)
+    if not err is None:
+        return err, 400
+    length, err = (None, None) if length_param is None else try_parse_int("length", length_param)
+    if not err is None:
+        return err, 400
+    if not length is None and length < -1:
+        return f"Negative length provided: {length}.\n", 400
+    if not offset is None and offset < -1:
+        return f"Negative offset provided: {offset}.\n", 400
+    if not os.path.exists(path):
+        return "", 404
+    if os.path.isdir(path):
+        return "Cannot read a directory.\n", 400
+    if offset is None or offset == -1:
+        return jsonify({
+            "data": "",
+            "offset": os.path.getsize(path),
+        })
+    f = open(path)
+    f.seek(offset)
+    max_length = int(os.getenv('COOK_FILE_SERVER_MAX_READ_LENGTH', '25000000'))
+    if length > max_length:
+        return f"Requested length for file read, {length} is greater than max allowed length, {max_length}", 400
+    data = f.read(max_length if length is None or length == -1 else length)
+    f.close()
+    return jsonify({
+        "data": data,
+        "offset": offset,
+    })
+
+
+permission_strings = {
+    '0': '---',
+    '1': '--x',
+    '2': '-w-',
+    '3': '-wx',
+    '4': 'r--',
+    '5': 'r-x',
+    '6': 'rw-',
+    '7': 'rwx'
```

Comment:
```suggestion
    '7': 'rwx',
```
You used trailing commas above